packages/kbn-dependency-usage/src/cli.ts

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import fs from 'fs';
+import nodePath from 'path';
+import yargs from 'yargs';
+import chalk from 'chalk';
+import madge from 'madge';
+
+import { groupFilesByOwners } from './lib/group_by_owners.ts';
+import { groupByPackage } from './lib/group_by_package.ts';
+
+interface CLiArgs {
+  dependencyName: string;
+  paths: string[];
+  groupBy: string;
+  summary: boolean;
+  filter: boolean;
+  outputPath: string;
+  collapseDepth: number;
+  tool: string;
+}
+
+function createCollapseRegexWithDepth(basePath: string, depth: number) {
+  let regex = `^${basePath}`;
+
+  for (let i = 0; i < depth; i++) {
+    regex += `/([^/]+)`;
+  }
+
+  return regex;
+}
+
+const invokeCruise = async ({
```

Comment:
I'm seeing strange behavior when running with cruise. It appears that namespaced dependencies do not fully resolve.
e.g.
```sh
./scripts/dependency-usage.sh -p x-pack/plugins/spaces --collapse-depth 2
Searching for dependencies in paths: x-pack/plugins/spaces
Dependencies will be collapsed to depth: 2
cruiser is used for building dependency graph
No output file specified; displaying results below:

{
  "x-pack/plugins/spaces/common": [
    "base64-js",
    "lodash",
    "rxjs"
  ],
  "x-pack/plugins/spaces/public": [
    "@emotion",
    "@hapi",
    "@testing-library",
    "enzyme",
    "lodash",
    "moment",
    "react",
    "react-dom",
    "react-router-dom",
    "react-use",
    "rxjs"
  ],
  "x-pack/plugins/spaces/server": [
    "@hapi",
    "lodash",
    "rxjs"
  ]
}
```

In this output we are reporting that `@hapi` is used, but `@hapi` is just a namespace for a bunch of different dependencies. Based on a search of the `spaces` plugin, I would have expected `@hapi/boom` to have been reported.

`madge` seems to get this right, but curiously includes an `undefined` entry at this collapse-depth:
```sh
./scripts/dependency-usage.sh -p x-pack/plugins/spaces --collapse-depth 2 --tool madge
Searching for dependencies in paths: x-pack/plugins/spaces
Dependencies will be collapsed to depth: 2
madge is used for building dependency graph
No output file specified; displaying results below:

{
  "common/": [
    "@types/lodash",
    "base64-js",
    "rxjs"
  ],
  "undefined": [],
  "public/": [
    "rxjs",
    "@types/enzyme",
    "@types/react",
    "@hapi/boom",
    "@testing-library/react",
    "@types/lodash",
    "moment",
    "react-use",
    "@types/react-dom",
    "@testing-library/user-event",
    "@testing-library/react-hooks",
    "@types/react-router-dom",
    "@emotion/react"
  ],
  "server/": [
    "@types/lodash",
    "rxjs",
    "@hapi/boom",
    "query-string"
  ]
}
```

There is also some disagreement between the two tools with `lodash`. `madge` thinks that we're only using the `@types/lodash`, while `cruise` correctly reports us using the concrete `lodash` package.

Reply:
It's because I've added the collapse rule for node_modules
```
 const collapseByNodeModuleRegex = '^node_modules/([^/]+)';
```

the output without it will be even more granular
```
  "x-pack/plugins/spaces": [
    "@emotion/react/dist/emotion-react.cjs.js",
    "@hapi/boom/lib/index.js",
    "@testing-library/react-hooks/lib/index.js",
    "@testing-library/react/dist/index.js",
    "@testing-library/user-event/dist/cjs/index.js",
    "base64-js/index.js",
    "enzyme/build/index.js",
    "lodash/lodash.js",
    "moment/moment.js",
    "react-dom/index.js",
    "react-dom/test-utils.js",
    "react-router-dom/index.js",
    "react-use/lib/useAsync.js",
    "react-use/lib/useAsyncFn.js",
    "react/index.js",
    "rxjs/dist/cjs/index.js"
  ]
```

Probably need to adjust that part for collapsing it better

Reply:
Nice catch, fixed in 43b5e0379c20fc024bc02f8a145b2308e2dcd9f0